pkg/database/db_seed.go

```diff
@@ -148,228 +192,518 @@ func seedDefaultTags() map[string]models.Tag {
-				var originalTag models.Tag
-				newTag.Type = models.TagType(tag.Kind)
-				newTag.Value = tag.Value
-
-				r := DB.Preload("HelpTopics").Where("type = ? AND value = ?", models.TagType(newTag.Type), newTag.Value).Find(&originalTag)
-				if r.Error != nil {
-					fmt.Println("Error: ", r.Error.Error())
-				} else if r.RowsAffected == 0 {
-					DB.Create(&newTag)
-					originalTag = newTag
-				}
-				// Clear all tags associations
-				err := DB.Model(&originalTag).Association("HelpTopics").Clear()
-				if err != nil {
-					fmt.Println("Failed clearing tags associations", err.Error())
+			if r.RowsAffected == 0 {
+				// Create new tag
+				if err := DB.Create(&newTag).Error; err != nil {
+					errMsg := fmt.Sprintf("failed to create tag %s=%s for help topic: %v", tag.Kind, tag.Value, err)
+					logrus.Error(errMsg)
+					result.addError(errMsg)
+					continue
 				}
+				originalTag = newTag
+				result.TagsCreated++
+			}
 
-				// Create tags help topic associations
-				err = DB.Model(&originalTag).Association("HelpTopics").Append(&helpTopic)
-				if err != nil {
-					fmt.Println("Failed creating tags associations", err.Error())
-				}
+			// Create tag-help topic association
+			if err := DB.Model(&originalTag).Association("HelpTopics").Append(&helpTopic); err != nil {
+				errMsg := fmt.Sprintf("failed to create tag association %s=%s for help topic %s: %v", tag.Kind, tag.Value, helpTopic.Name, err)
+				logrus.Error(errMsg)
+				result.addError(errMsg)
+				continue
+			}
+
+			helpTopic.Tags = append(helpTopic.Tags, originalTag)
 
-				DB.Save(&originalTag)
+			if err := DB.Save(&helpTopic).Error; err != nil {
+				errMsg := fmt.Sprintf("failed to save help topic %s after adding tag: %v", helpTopic.Name, err)
+				logrus.Error(errMsg)
+				result.addError(errMsg)
+				continue
+			}
+
+			if err := DB.Save(&originalTag).Error; err != nil {
+				errMsg := fmt.Sprintf("failed to save tag %s=%s for help topic: %v", tag.Kind, tag.Value, err)
+				logrus.Error(errMsg)
+				result.addError(errMsg)
+				continue
 			}
 		}
 	}
+	return nil
+}
+
+// logSeedingResults logs the final results of the seeding process
+func logSeedingResults(result *SeedingResult) {
+	logrus.Infof("=== SEEDING PROCESS COMPLETE ===")
+	logrus.Infof("Quickstarts: %d processed (%d created, %d updated)",
+		result.QuickstartsProcessed, result.QuickstartsCreated, result.QuickstartsUpdated)
+	logrus.Infof("Help Topics: %d processed (%d created, %d updated)",
+		result.HelpTopicsProcessed, result.HelpTopicsCreated, result.HelpTopicsUpdated)
+	logrus.Infof("Tags: %d created", result.TagsCreated)
+	logrus.Infof("Favorites: %d restored", result.FavoritesRestored)
+
+	if len(result.Errors) > 0 {
+		logrus.Errorf("Completed with %d errors", len(result.Errors))
+		for i, err := range result.Errors {
+			logrus.Errorf("Error %d: %v", i+1, err)
+		}
+	} else {
+		logrus.Info("Completed successfully with no errors")
+	}
+}
 
-	SeedFavorites(favorites)
+func SeedTags() error {
```

Comment:
Is `SeedTags` appropriate for this func? It seems to be doing more than just tags